watertap/unit_models/electrodialysis_1D.py

```diff
@@ -559,50 +655,159 @@ def eq_isothermal_concentrate(self, t, x):
                     == self.current_applied[t]
                 )
             else:
-                return (
-                    self.current_density_x[t, x] * self.total_areal_resistance_x[t, x]
-                    == self.voltage_applied[t]
-                )
+                if self.config.has_nonohmic_potential_membrane:
+                    if self.config.has_Nernst_diffusion_layer:
+                        return (
+                            self.current_density_x[t, x]
+                            * self.total_areal_resistance_x[t, x]
+                            + (
+                                self.potential_ohm_dl_x["cem", t, x]
+                                + self.potential_ohm_dl_x["aem", t, x]
+                                + self.potential_nonohm_dl_x["cem", t, x]
+                                + self.potential_nonohm_dl_x["aem", t, x]
+                                + self.potential_nonohm_membrane_x["cem", t, x]
+                                + self.potential_nonohm_membrane_x["aem", t, x]
+                            )
+                            * self.cell_pair_num
+                            == self.voltage_applied[t]
+                        )
+                    else:
+                        return (
+                            self.current_density_x[t, x]
+                            * self.total_areal_resistance_x[t, x]
+                            + (
+                                self.potential_nonohm_membrane_x["cem", t, x]
+                                + self.potential_nonohm_membrane_x["aem", t, x]
+                            )
+                            * self.cell_pair_num
+                            == self.voltage_applied[t]
+                        )
+                else:
+                    if self.config.has_Nernst_diffusion_layer:
+                        return (
+                            self.current_density_x[t, x]
+                            * self.total_areal_resistance_x[t, x]
+                            + (
+                                self.potential_ohm_dl_x["cem", t, x]
+                                + self.potential_ohm_dl_x["aem", t, x]
+                                + self.potential_nonohm_dl_x["cem", t, x]
+                                + self.potential_nonohm_dl_x["aem", t, x]
+                            )
+                            * self.cell_pair_num
+                            == self.voltage_applied[t]
+                        )
+                    else:
+                        return (
+                            self.current_density_x[t, x]
+                            * self.total_areal_resistance_x[t, x]
+                            == self.voltage_applied[t]
+                        )
 
         @self.Constraint(
             self.flowsheet().time,
             self.diluate.length_domain,
             doc="calcualte length_indexed voltage",
         )
         def eq_get_voltage_x(self, t, x):
```

Comment:
May be I am confused here, why is this constraint for "eq_get_voltage_x" repeated same as "eq_get_current_density" ?

Reply:
The vary in the right-hand side of the equation as `voltage_applied` or `voltage_x`.  This constraint is essentially for getting the x-dependent voltage under the Const_Curr mode, while it simply says `v_x == v_applied`  in the other mode.  So this single constraint works for both modes.